### PR TITLE
Patch for working with Mailgun regions

### PR DESCRIPTION
**Pull request by alvisx**
## Proposed changes

Mailgun added opportunity of region selection in control panel. Currently available 2 regions: US and EU. Domains added in EU region is not visible in main API (US) because base URL for EU region is like _https://api.eu.mailgun.net/v3/<domain>_. Mailgun driver in Mail provider works wrong if domain situated in EU region. 

## Types of changes
 - Region defining in _.env_ 
 - Defining default region in _<app_root>/config/mail.js_
 - Building url with region in _src/Mail/Drivers/Mailgun.js_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/adonisjs/adonis-mail/blob/develop/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

## Further comments

No

### CODE REVIEW

**Comment by RomainLanz** [comment]
Hey @alvisx! 👋 

Could you please update your message and fill the placeholder? 😃

File: src/Mail/Drivers/Mailgun.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ class MailGunTransporter {
    * @return {String}
    */
   get endpoint () {
-    return `https://api.mailgun.net/v3/${this.config.domain}/messages.mime`
+    return `https://api.${this.config.region.length > 0 ? this.config.region.toLowerCase() + '.' : ''}mailgun.net/v3/${this.config.domain}/messages.mime`
```
Comment (thetutlage) [inline src/Mail/Drivers/Mailgun.js]: I believe this can be handled in a better way, knowing the fact that `region` can be `undefined` as well.

```js
const urlParts = ['api']
if (typeof (this.config.region) === 'string') {
  urlParts.push(this.config.region.toLowerCase)
}

https://${urlParts.join('.')}mailgun.net/v3/${this.config.domain}/messages.mime
```

Comment (alvisx) [reply src/Mail/Drivers/Mailgun.js]: Hello, it really looks better. Sorry for delay with answer.

**Comment by alvisx** [comment]
My next steps is writing tests or update PR [this changes](https://github.com/adonisjs/adonis-mail/pull/31#discussion_r263232573)?

**Comment by thetutlage** [comment]
Yes, tests will help that we aren't missing any edge cases

**Comment by RomainLanz** [comment]
Closing this PR since a similar one has been merged.

Reference: https://github.com/adonisjs/adonis-mail/pull/33
